modules/platform/nuxeo-drive-server/nuxeo-drive-mongodb/src/test/java/org/nuxeo/drive/test/NuxeoDriveMongoDBAuditFeature.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.nuxeo.drive.test;
+
+import java.util.concurrent.TimeUnit;
+
+import org.nuxeo.drive.mongodb.TestMongoDBAuditChangeFinder;
+import org.nuxeo.ecm.automation.test.AutomationFeature;
+import org.nuxeo.ecm.platform.audit.AuditFeature;
+import org.nuxeo.ecm.platform.audit.api.AuditLogger;
+import org.nuxeo.mongodb.audit.MongoDBAuditFeature;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.RunnerFeature;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+/**
+ * MongoDB audit feature for nuxeo drive. This Feature is necessary for the deployment annotations to be taken into
+ * account by {@link TestMongoDBAuditChangeFinder}.
+ *
+ * @since 11.2
+ */
+@Features({ AutomationFeature.class, AuditFeature.class, MongoDBAuditFeature.class })
+@Deploy("org.nuxeo.drive.mongodb")
+public class NuxeoDriveMongoDBAuditFeature implements RunnerFeature {
+
+    @Override
+    public void initialize(FeaturesRunner runner) {
+        runner.getFeature(TransactionalFeature.class)
+              .addWaiter(duration -> {
+                  // Wait for audit completion
+                  return Framework.getService(AuditLogger.class).await(duration.toMillis(), TimeUnit.MILLISECONDS);
+              });
+    }
```

Comment:
You don't need this since it will now be handled by the `MongoDBAuditFeature`.
Then, of course, we'll have to validate that this change in `MongoDBAuditFeature` doesn't break other tests using it.
Though the T&P will validate it, you can already run locally the 6 tests using it in `nuxeo-audit-storage-directory` and `nuxeo-platform-audit-mongodb`.

Reply:
org.nuxeo.audit.storage.impl.TestRestorationFromDirectoryAuditStorage is affected by the initialize overriding:
The test takes 400 seconds instead of 40 and gives a lot of errors while trying and retrying. Sometimes eclipse marks it as failing sometimes succeeding. Here the exceptions I get:
```
2020-08-05 15:05:50,208 [audit/writerPool-00,in:1,inCheckpoint:0,out:0,lastRead:1596632627011,lastTimer:1596632626942,wm:209273831663206400,loop:504,record] WARN  [AbstractBatchComputation] Computation: audit/writer fails to process batch of 1 records, last record: audit-audit-00:+79366700662784, retrying ...
com.mongodb.MongoBulkWriteException: Bulk write operation error on server localhost:27017. Write errors: [BulkWriteError{index=0, code=11000, message='E11000 duplicate key error collection: unittests.audit index: _id_ dup key: { : 7 }', details={}}]. 
	at com.mongodb.connection.BulkWriteBatchCombiner.getError(BulkWriteBatchCombiner.java:173) ~[mongo-java-driver-3.12.1.jar:?]
```
Other tests are fine

Reply:
OK, strange. So you can keep th code in the `NuxeoDriveMongoDBAuditFeature`.
Hoping the Drive tests will pass.

Reply:
Not yet 😭